pipelines/batch/src/main/java/org/openmrs/analytics/JdbcFetchUtil.java

```diff
@@ -137,6 +141,62 @@ public void processElement(@Element KV<String, Iterable<String>> element,
 		}
 	}
 	
+	/**
+	 * Utilizes Beam JdbcIO to query for resources directly from FHIR (HAPI) server's database and
+	 * returns a PCollection of Lists of String objects - each corresponding to a resource's payload
+	 */
+	public static class FetchRowsJdbcIo extends PTransform<PCollection<List<String>>, PCollection<List<String>>> {
+		
+		private final JdbcIO.DataSourceConfiguration dataSourceConfig;
+		
+		public FetchRowsJdbcIo(JdbcIO.DataSourceConfiguration dataSourceConfig) {
+			this.dataSourceConfig = dataSourceConfig;
+		}
+		
+		@Override
+		public PCollection<List<String>> expand(PCollection<List<String>> queryParameters) {
+			return queryParameters.apply("JdbcIO readAll",
+			    JdbcIO.<List<String>, List<String>> readAll().withDataSourceConfiguration(dataSourceConfig)
+			            .withCoder(ListCoder.of(StringUtf8Coder.of()))
+			            .withParameterSetter(new JdbcIO.PreparedStatementSetter<List<String>>() {
+				            
+				            @Override
+				            public void setParameters(List<String> element, PreparedStatement preparedStatement)
+				                    throws Exception {
+					            preparedStatement.setString(1, element.get(0));
+					            preparedStatement.setInt(2, Integer.valueOf(element.get(1)));
+					            preparedStatement.setInt(3, Integer.valueOf(element.get(2)));
+				            }
+			            }).withOutputParallelization(true)
+			            .withQuery(
+			                "SELECT res.res_id, res.res_type, res.res_updated, res.res_ver, ver.res_encoding, ver.res_text FROM hfj_resource res, hfj_res_ver ver WHERE res.res_type=? AND res.res_id = ver.res_id AND res.res_id %? = ?")
```

Comment:
Consider breaking this long string into multiple lines, both for style compliance and better readability.